server/middleware/oidc/index.js

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) 2024 Uber Technologies Inc.
+//
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+const OpenIDClient = require('openid-client');
+const passport = require('koa-passport');
+const session = require('koa-session');
+const allowUrl = ['/login', '/logout', '/oauth2/redirect'];
+
+let settings = {};
+
+try {
+  settings = require('../../config/oidc');
+} catch (e) {
+  console.log('OIDC configuration file not found, using ENV variables');
+}
+
+const clientID = settings.clientID || process.env.OPENID_CLIENT_ID;
+const clientSecret = settings.clientSecret || process.env.OPENID_CLIENT_SECRET;
+const callbackURL = settings.callbackURL || process.env.OPENID_CALLBACK_URL;
+const discoverURL = settings.discoverURL || process.env.OPENID_DISCOVER_URL;
+const scope = settings.scope || process.env.OPENID_SCOPE || 'openid';
+
+const middleware = async function(ctx, next) {
```

Comment:
After investigation, few things to confirm:
- The [middleware for sending admin jwt](https://github.com/uber/cadence-web/blob/517328a5f50e1570f628813088dff1d7ed03298b/server/index.js#L118) is no longer useful as the new implementation you've provided on `cadence-frontend` doesn't use it.
- The default implementation is now using OIDC

Reply:
Those two plugins are meant for different use cases: If you are managing your own certificates and you are good with use "admin" level requests for everyone, you can use admin jwt as a middleware. 
On the other hand, if you wish to have group based permissions and/or you delegate identity provider to 3rd party, you can use oidc middleware instead of "admin jwt" middleware

Reply:
Both cases do work.
If it is an Admin JWT - it will be allowed: https://github.com/uber/cadence/blob/master/common/authorization/oauthAuthorizer.go#L150
If not - you can have fine-tuned auth per domain https://github.com/uber/cadence/blob/master/common/authorization/oauthAuthorizer.go#L159

However, Uber's implementation does not rely on this header or use this logic.
So this is only for oAuth support. Not sure if we have an integration test for this case.